paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/gpay/GooglePayConfirmationActivityTest.kt

```diff
@@ -0,0 +1,362 @@
+
+            intendingGooglePayToBeLaunched(GooglePayPaymentMethodLauncher.Result.Completed(paymentMethod))
+            intendingPaymentConfirmationToBeLaunched(InternalPaymentResult.Canceled)
+
+            confirmationHandler.start(
+                ConfirmationHandler.Args(
+                    intent = PAYMENT_INTENT,
+                    confirmationOption = GOOGLE_PAY_CONFIRMATION_OPTION,
+                )
+            )
+
+            val confirmingWithGooglePay = awaitItem().assertConfirming()
+
+            assertThat(confirmingWithGooglePay.option).isEqualTo(GOOGLE_PAY_CONFIRMATION_OPTION)
+
+            intendedGooglePayToBeLaunched()
+
+            val confirmingWithSavedPaymentMethod = awaitItem().assertConfirming()
+
+            assertThat(confirmingWithSavedPaymentMethod.option)
+                .isEqualTo(
+                    PaymentMethodConfirmationOption.Saved(
+                        initializationMode = GOOGLE_PAY_CONFIRMATION_OPTION.initializationMode,
+                        shippingDetails = GOOGLE_PAY_CONFIRMATION_OPTION.shippingDetails,
+                        paymentMethod = paymentMethod,
+                        optionsParams = null,
+                    )
+                )
+
+            intendedPaymentConfirmationToBeLaunched()
+
+            val canceledAction = awaitItem().assertComplete().result.assertCanceled()
+
+            assertThat(canceledAction.action).isEqualTo(ConfirmationHandler.Result.Canceled.Action.InformCancellation)
+        }
+    }
+
+    private fun test(
+        test: suspend ConfirmationActivity.() -> Unit
+    ) = runTest(UnconfinedTestDispatcher()) {
+        val countDownLatch = CountDownLatch(1)
+
+        ActivityScenario.launch<ConfirmationActivity>(
+            Intent(application, ConfirmationActivity::class.java)
+        ).use { scenario ->
+            scenario.onActivity { activity ->
+                launch {
+                    test(activity)
+                    countDownLatch.countDown()
+                }
+            }
+
+            countDownLatch.await(10, TimeUnit.SECONDS)
+        }
+    }
+
+    private fun intendingGooglePayToBeLaunched(result: GooglePayPaymentMethodLauncher.Result) {
+        intending(hasComponent(GOOGLE_PAY_ACTIVITY_NAME)).respondWith(
+            Instrumentation.ActivityResult(
+                Activity.RESULT_OK,
+                Intent().putExtra("extra_result", result)
+            )
+        )
+    }
+
+    private fun intendingPaymentConfirmationToBeLaunched(result: InternalPaymentResult) {
+        intending(hasComponent(PAYMENT_CONFIRMATION_LAUNCHER_ACTIVITY_NAME)).respondWith(
+            Instrumentation.ActivityResult(
+                Activity.RESULT_OK,
+                Intent().putExtras(bundleOf("extra_args" to result))
+            )
+        )
+    }
+
+    private fun intendedGooglePayToBeLaunched() {
+        intended(hasComponent(GOOGLE_PAY_ACTIVITY_NAME))
+    }
```

Comment:
It is actually possible to validate the arguments passed to the activity as well but we need to make the contract arguments library public to use them. Any objections to doing this? I can follow-up with it and additional tests.